pennylane_qrack/qrack_device.py

```diff
@@ -199,36 +256,52 @@ def _apply_basis_state(self, op):
     def _apply_gate(self, op):
         """Apply native qrack gate"""
 
-        # translate op wire labels to consecutive wire labels used by the device
-        device_wires = self.map_wires(op.wires)
-        par = op.parameters
-
         opname = op.name
         if isinstance(op, Adjoint):
             op = op.base
             opname = op.name + ".inv"
 
-        if opname == "Toffoli" or opname == "Toffoli.inv" or opname == "CNOT" or opname == "CNOT.inv" or opname == "MultiControlledX" or opname == "MultiControlledX.inv":
+        if opname == "MultiRZ":
+            device_wires = self.map_wires(op.wires)
+            for q in device_wires:
+                self._state.r(Pauli.PauliZ, par[0], q)
+        elif opname == "C(MultiRZ)":
+            device_wires = self.map_wires(op.wires)
+            control_wires = self.map_wires(op.control_wires)
+            for q in device_wires:
+                self._state.mcr(Pauli.PauliZ, par[0], control_wires, q)
+
+        # translate op wire labels to consecutive wire labels used by the device
+        device_wires = self.map_wires((op.control_wires + op.wires) if op.control_wires else op.wires)
+        par = op.parameters
+
+        if opname in ["Toffoli", "Toffoli.inv", "C(Toffoli)", "C(Toffoli).inv", "CNOT", "CNOT.inv", "C(CNOT)", "C(CNOT).inv", "MultiControlledX", "MultiControlledX.inv", "C(PauliX)", "C(PauliX).inv"]:
```

Comment:
my suggestion is to use product here:
```
from itertools import product

[''.join(p) for p in product(['Toffoli', 'CNOT', etc...], ['', '.inv'])]
```
Not much to shorten it up, but rather to make sure you don't forget the `.inv` versions.

Reply:
Done.